src/library_nodefs.js

```diff
@@ -235,7 +235,7 @@ addToLibrary({
         var path = NODEFS.realPath(node);
         try {
           path = fs.readlinkSync(path);
-          path = nodePath.relative(nodePath.resolve(node.mount.opts.root), path);
+          path = nodePath.relative( PATH.isAbs(path) ? NODEFS.realPath(node.parent) : nodePath.resolve(node.mount.opts.root), path );
```

Comment:
I wonder if we should even be calling `nodePath.relative` at all?  What happens if you just remove this line?

Reply:
Without this line, the `resolve` method which works correctly with relative paths will return wrong paths when using absolute paths. The idea with this line is to return a relative path no matter what.

Reply:
But is that how readlink is supposed to work?    If the link as absolute then shouldn't readlink return an absolute path?

Reply:
@sbc100 You're correct. Removing this line resolved the issues I was experiencing.

Reply:
It looks like issue #3222 was the reason for this line.

Reply:
Hmm.. from my reading of #3222 and the fix that landed with the test_unistd_symlink_on_nodefs test, i don't think it is correct. See `https://github.com/emscripten-core/emscripten/pull/21805#issuecomment-2094926239` below